### DISCUSSION TITLE
Add AVIF and HEIF/HEIC image file formats to convert options

### DISCUSSION THREAD

**Discussion by k1logr4m** [deckerst/aves#908] [Ideas]
I've been playing around with recent file formats. I was happy to know that I can convert to webp images with Aves and thought it would be cool to have AVIF and HEIF/HEIC among the options.

**Comment by deckerst** [comment]
I could take a look for AVIF encoding on Android 14+, but an important thing to bear in mind is that Aves does not support AVIF metadata editing, so you would lose metadata when converting to AVIF. Losing date, location, tags, etc.

**Reply by k1logr4m** [reply]
I will keep that in mind. Thanks!

**Comment by kmcbest** [comment]
> I could take a look for AVIF encoding on Android 14+, but an important thing to bear in mind is that Aves does not support AVIF metadata editing, so you would lose metadata when converting to AVIF. Losing date, location, tags, etc.

Since AVIF seems like to have won the battle and becomes the most supported new-generation image format, how about implement its full support (conversion to and from, meta-editing, etc) in Aves?

**Reply by deckerst** [reply]
I delegate that metadata reading and editing to libraries, as there's already plenty enough to do at a higher level in Aves. If AVIF gets to actually be used, support will naturally come from these libraries (or Android itself).

**Reply by kmcbest** [reply]
> I delegate that metadata reading and editing to libraries, as there's already plenty enough to do at a higher level in Aves. If AVIF gets to actually be used, support will naturally come from these libraries (or Android itself).

Is it related to this upstream issue? https://github.com/flutter/flutter/issues/61229

**Reply by deckerst** [reply]
No, that issue is about decoding at flutter level. Metadata editing is something else entirely. Flutter wouldn't care about metadata for any format.

**Reply by kmcbest** [reply]
OK so what are the libraries you mentioned, which have a say about metadata? I want to keep an eye on their progress.

**Reply by deckerst** [reply]
Aves mainly uses [ExifInterface](https://developer.android.com/jetpack/androidx/releases/exifinterface) and [PixyMeta](https://github.com/deckerst/pixymeta-android).

**Reply by kmcbest** [reply]
So if I understand correctly. Aves will depend on Flutter's update to be able to convert to and from AVIF, then ExifInterface and PixyMeta's update to be able to retain Metadata during that conversion?

--
Edit: It seems ExifInterface [has already supported AVIF] in November 2024 (https://developer.android.com/jetpack/androidx/releases/exifinterface#1.4.0-alpha01)

**Reply by deckerst** [reply]
> Aves will depend on Flutter's update to be able to convert to and from AVIF

No, Aves uses native Android decoders for AVIF decoding (and possibly encoding, but this is not implemented). Decoding already works on recent devices. 

> ExifInterface and PixyMeta's update to be able to retain Metadata during that conversion?

Possibly.

> It seems ExifInterface [has already supported AVIF] in November 2024

Yes, but in practice it doesn't seem to work. At least when I tried on my sample AVIF files some time ago.
